cores/esp8266/core_esp8266_wiring_pulse.cpp

```diff
@@ -43,10 +50,16 @@ unsigned long pulseIn(uint8_t pin, uint8_t state, unsigned long timeout)
     }
     const uint32_t timeout_cycles = microsecondsToClockCycles(timeout);
     const uint32_t start_cycle_count = xthal_get_ccount();
-    WAIT_FOR_PIN_STATE(!state);
-    WAIT_FOR_PIN_STATE(state);
+    if (!waitForPinState(pin, !state, timeout_cycles, start_cycle_count)) {
+        return 0;
+    }
+    if (!waitForPinState(pin, state, timeout_cycles, start_cycle_count)) {
+        return 0;
+    }
     const uint32_t pulse_start_cycle_count = xthal_get_ccount();
-    WAIT_FOR_PIN_STATE(!state);
+    if (!waitForPinState(pin, !state, timeout_cycles, start_cycle_count)) {
+        return 0;
```

Comment:
This is wrong per my reading of the (not so hot docs)  https://www.arduino.cc/reference/en/language/functions/advanced-io/pulsein/?setlang=it

Return 0 on no pulse beginning edge detected.  OTW, it should return the full timeout as the pulse width  (i.e. get rid of the `return 0` and just fallthru tho the return below).

Reply:
@earlephilhower Can you agree this year; the change is just same effective execution path as with the macros before?